contracts/TranscodeJobs.sol

```diff
@@ -0,0 +1,152 @@
+        // Check if job is active
+        if (jobStatus(self, _jobId) != JobStatus.Active) throw;
+        // Check if sender is assigned transcoder
+        if (self.jobs[_jobId].transcoderAddress != msg.sender) throw;
+        // Check if previous verification period over
+        if (block.number < self.jobs[_jobId].endVerificationBlock) throw;
+
+        self.jobs[_jobId].lastClaimedSegmentRange[0] = _startSegmentSequenceNumber;
+        self.jobs[_jobId].lastClaimedSegmentRange[1] = _endSegmentSequenceNumber;
+        self.jobs[_jobId].lastClaimedWorkBlock = block.number;
+        self.jobs[_jobId].endVerificationBlock = block.number + _verificationPeriod;
+        self.jobs[_jobId].transcodeClaimsRoots.push(_transcodeClaimsRoot);
+
+        return true;
+    }
+
+    function verify(Jobs storage self, uint256 _jobId, uint256 _segmentSequenceNumber, bytes32 _dataHash, bytes32 _transcodedDataHash, bytes _broadcasterSig, bytes _proof, uint64 _verificationRate) constant returns (bool) {
+        // Check if job is active
+        if (jobStatus(self, _jobId) != JobStatus.Active) return false;
+        // Check if sender is the assigned transcoder
+        if (self.jobs[_jobId].transcoderAddress != msg.sender) return false;
+        // Check if segment is eligible for verification
+        if (!shouldVerifySegment(_segmentSequenceNumber,
+                                 self.jobs[_jobId].lastClaimedSegmentRange[0],
+                                 self.jobs[_jobId].lastClaimedSegmentRange[1],
+                                 self.jobs[_jobId].lastClaimedWorkBlock,
+                                 block.blockhash(self.jobs[_jobId].lastClaimedWorkBlock),
+                                 _verificationRate
+                                 )) return false;
+        // Check if segment was signed by broadcaster
+        if (!ECVerify.ecverify(sha3(self.jobs[_jobId].streamId, _segmentSequenceNumber, _dataHash), _broadcasterSig, self.jobs[_jobId].broadcasterAddress)) return false;
+        // Check if transcode claim is included in the Merkle root submitted during the last call to claimWork()
+        if (!MerkleProof.verifyProof(_proof,
+                                     self.jobs[_jobId].transcodeClaimsRoots[self.jobs[_jobId].transcodeClaimsRoots.length - 1],
+                                     sha3(self.jobs[_jobId].streamId, _segmentSequenceNumber, _dataHash, _transcodedDataHash, _broadcasterSig))) return false;
+
+        return true;
+    }
+
+    /*
+     * Computes whether a segment is eligible for verification based on the last call to claimWork()
+     * @param _jobId Job identifier
+     * @param _segmentSequenceNumber Sequence number of segment in stream
+     * @param _verificationRate Rate at which a particular segment should be verified
+     */
+    function shouldVerifySegment(uint256 _segmentSequenceNumber,
```

Comment:
This helper function determines if a segment should be verified or not. Rather than access state (i.e. a `Jobs` struct), all dependencies are passed in as parameters in order to make this function easier to test.